### PR TITLE
feat(datafusion): add case and if-else statements

### PR DESCRIPTION
**Pull request by mesejo**

### CODE REVIEW

File: ibis/backends/datafusion/compiler/values.py [hunk 1]
```diff
@@ -670,3 +671,19 @@ def _window_boundary(op, *, value, preceding, **_):
...
+
+
+@translate_val.register(ops.IfElse)
+def _if_else(op, *, bool_expr, true_expr, false_null_expr, **_):
+    return sg.exp.Case().when(bool_expr, true_expr).else_(false_null_expr)
+
+
+@translate_val.register(ops.NotNull)
+def _not_null(op, *, arg, **_):
+    return sg.exp.Not(this=sg.exp.Is(this=arg, expression=NULL))
```
Comment (cpcloud) [inline ibis/backends/datafusion/compiler/values.py:689]: You should be able to use `sg.not_(arg.is_(NULL))` here.

File: ibis/backends/datafusion/compiler/values.py [hunk 2]
```diff
@@ -670,3 +671,19 @@ def _window_boundary(op, *, value, preceding, **_):
...
+
+@translate_val.register(ops.SimpleCase)
+@translate_val.register(ops.SearchedCase)
+def _case(op, *, base=None, cases, results, default, **_):
+    return sg.exp.Case(this=base, ifs=list(map(if_, cases, results)), default=default)
+
+
+@translate_val.register(ops.IfElse)
+def _if_else(op, *, bool_expr, true_expr, false_null_expr, **_):
+    return sg.exp.Case().when(bool_expr, true_expr).else_(false_null_expr)
```
Comment (cpcloud) [inline ibis/backends/datafusion/compiler/values.py:684]: Does `if_(bool_expr, true_expr, false_expr)` work here?

**Review by cpcloud — APPROVED** [review]
Couple of style comments. Nothing blocking.

File: ibis/backends/datafusion/compiler/values.py (hunk 2 above)
Comment (mesejo) [reply ibis/backends/datafusion/compiler/values.py]: Yes, it does 😄

**Pull request merged**